### ISSUE TITLE
Does not find cuDNN library

### ISSUE THREAD

**Issue by ccoulombe** [dmlc/dgl#647]
## 🐛 Bug

On an HPC system, when building DGL with `cmake -DUSE_CUDA ..` as stated in the docs, configuration does not found cuDNN library which may not live in `CUDA_TOOLKIT_ROOT_DIR` .

```
-- Found CUDA_CUDNN_LIBRARY=CUDA_CUDNN_LIBRARY-NOTFOUND
```

## To Reproduce

Build DGL from source on a system where `cuDNN` is not in [`${CUDA_TOOLKIT_ROOT_DIR}`](https://github.com/dmlc/dgl/blob/e16e895d5364cdff243c2fb548e27fbc6366bf2b/cmake/util/FindCUDA.cmake#L55)

## Expected behavior

Find `cuDNN` when it is available via an environment variable such as : `CUDNN_LIB_DIR`

## Environment

 - DGL Version (e.g., 1.0): 0.2
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): torch
 - OS (e.g., Linux): Linux Centos
 - How you installed DGL (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): `cmake -DUSE_CUDA ..`
 - Python version: 3.7
 - CUDA/cuDNN version (if applicable): 10 and 7.5
 - GPU models and configuration (e.g. V100): NA
 - Any other relevant information:

## Side Note

Pytorch correctly finds `cuDNN` when building from source with an env. variable [`CUDNN_INCLUDE_DIR`](https://github.com/pytorch/pytorch/blob/b46e87cd3df1024bf287ad82857e57aa7a07a5aa/cmake/public/cuda.cmake#L123)

I'd propose to use the same solution.

**Comment by ccoulombe** [comment]
In the meantime, one can still build by specifying `cuDNN` lib directory 
```
cmake -DUSE_CUDA=ON -DCUDA_CUDNN_LIBRARY=<cudnn_lib_dir>/lib64 ..
```

**Comment by jermainewang** [comment]
We are not using CUDNN at all actually. Should skip CUDNN during compilation. Thank you for the report!

**Comment by BarclayII** [comment]
Should be fixed in #636
